lib/buffer.js

```diff
@@ -326,13 +326,26 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  // No need to verify that "this.length <= MAX_UINT32" since it's a read-only
+  // property of a typed array.
+
+  // This behaves neither like String nor Uint8Array in that we set start/end
+  // to their upper/lower bounds if the value passed is out of range.
+  if (start === undefined || start < 0)
+    start = 0;
+  // Return early if start > this.length. Done here to prevent potential uint32
+  // coercion fail below.
+  if (start > this.length)
+    return '';
+
+  if (end === undefined || end > this.length)
+    end = this.length;
+
+  // Force coersion to uint32. This will also coerce falsey/NaN values to 0.
+  if (end <= start || (end >>>= 0) <= (start >>>= 0))
```

Comment:
Ingenious way of doing this, though my gut tells me there's a better way. Instead of doing simultaneous assignments and checks in the same conditional.

@bnoordhuis you have a suggestion of a way around this?